### PR TITLE
Api doc update

### PR DESCRIPTION
**Pull request by pitr-ch**
see the commits` messages

### CODE REVIEW

**Comment by pitr-ch** [comment]
I've found a bug in this :( please wait until I fix it

**Comment by pitr-ch** [comment]
@ohadlevy fixed

File: app/controllers/api/base_controller.rb [hunk 1]
```diff
@@ -77,21 +87,18 @@ def deny_access(details = nil)
...
-                      when (id = params[:"#{resource_name}_id"]).present?
-                        ['id', id]
-                      when (name = params[:"#{resource_name}_name"]).present?
-                        ['name', name]
-                      else
-                        [nil, nil]
-                    end
-
-      resource = resource_class.send(:"find_by_#{finder}", key) if finder
+      possible_keys = %w(id name login)
```
Comment (ohadlevy) [inline app/controllers/api/base_controller.rb]: i know it was here before, but do we actually use login anywhere?

File: app/controllers/api/v1/config_templates_controller.rb [hunk 2]
```diff
@@ -8,24 +8,26 @@ class ConfigTemplatesController < V1::BaseController
 
       api :GET, "/config_templates/", "List templates"
       param :search, String, :desc => "filter results"
-      param :order,  String, :desc => "sort results"
+      param :order, String, :desc => "sort results"
       def index
         @config_templates = ConfigTemplate.search_for(params[:search], :order => params[:order]).
-          includes(:operatingsystems, :template_combinations, :template_kind)
+            includes(:operatingsystems, :template_combinations, :template_kind)
```
Comment (ohadlevy) [inline app/controllers/api/v1/config_templates_controller.rb:14]: whitespace

File: app/controllers/api/base_controller.rb (hunk 1 above)
Comment (pitr-ch) [inline app/controllers/api/base_controller.rb]: login is there for User model to be able to be queried by login

Comment (ohadlevy) [inline app/controllers/api/base_controller.rb]: but surely we dont need the login method in the api?

Comment (pitr-ch) [inline app/controllers/api/base_controller.rb]: It's not related to a login method. `possible_keys` are representing posible attributes which can be used to find a model object. It tries to call `.find_by_id`, `.find_by_name`, `.find_by_login` if the method is present.

**Comment by ohadlevy** [comment]
merged -thanks!

**Pull request closed without merging**